### PR TITLE
LOG-839: Remove unused fluent env vars

### PR DESCRIPTION
**Pull request by jcantrill**
This PR:
* Removes unused env vars
* Replaces env vars in config in favor of generating defaults where appropriate

The change of interest is the commti for LOG-839

Ref: https://issues.redhat.com/browse/LOG-839

### CODE REVIEW

**Comment by periklis** [comment]
@jcantrill After taking a closer look on this, I can attest that it is not conflicting with #596 except that both drop `BUFFER_QUEUE_FULL_ACTION`. But this is not a big deal.

**Comment by jcantrill** [comment]
> @jcantrill After taking a closer look on this, I can attest that it is not conflicting with #596 except that both drop `BUFFER_QUEUE_FULL_ACTION`. But this is not a big deal.

I need to push a change as the test log shows the value for this field is not a one of the valid enum values

**Comment by periklis** [comment]
> > @jcantrill After taking a closer look on this, I can attest that it is not conflicting with #596 except that both drop `BUFFER_QUEUE_FULL_ACTION`. But this is not a big deal.
> 
> I need to push a change as the test log shows the value for this field is not a one of the valid enum values

Yes the fluentdForward case is using `exception` and that should be `throw_exception`. Noticed the same on my tuning spec PR.

**Comment by jcantrill** [comment]
Blocked pending the outcome of https://github.com/openshift/cluster-logging-operator/pull/596  This PR may need to be significantly altered

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go [hunk 1]
```diff
@@ -316,32 +316,32 @@ var _ = Describe("Generating fluentd config", func() {
...
 				</filter>
 
 				<filter kubernetes.journal.**>
 					@type parse_json_field
-					merge_json_log "#{ENV['MERGE_JSON_LOG'] || 'false'}"
-					preserve_json_log "#{ENV['PRESERVE_JSON_LOG'] || 'true'}"
-					json_fields "#{ENV['JSON_FIELDS'] || 'MESSAGE,log'}"
+					merge_json_log 'false'
+					preserve_json_log 'true'
+					json_fields 'log,MESSAGE'
```
Comment (periklis) [inline pkg/generators/forwarding/fluentd/fluent_conf_test.go:330]: Is the order of `log` and `MESSAGE` important for the filter plugin?

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go [hunk 2]
```diff
@@ -950,11 +950,11 @@ var _ = Describe("Generating fluentd config", func() {
...
 						type_name _doc
 						write_operation create
-						reload_connections "#{ENV['ES_RELOAD_CONNECTIONS'] || 'true'}"
+						reload_connections 'true'
 						# https://github.com/uken/fluent-plugin-elasticsearch#reload-after
-						reload_after "#{ENV['ES_RELOAD_AFTER'] || '200'}"
+						reload_after '200'
 						# https://github.com/uken/fluent-plugin-elasticsearch#sniffer-class-name
-						sniffer_class_name "#{ENV['ES_SNIFFER_CLASS_NAME'] || 'Fluent::Plugin::ElasticsearchSimpleSniffer'}"
+
```
Comment (periklis) [inline pkg/generators/forwarding/fluentd/fluent_conf_test.go:957]: Is the `sniffer_class_name` drop intentionally?

**Comment by jcantrill** [comment]
> LGTM. Besides the question below, why are on some places boolean, number and reserved words value quoted with single quotes and on other not?, e.g. `'false', '-1', 'nil'`

I replaced the values as they would have been before with the env evaluation.  In the interest of not breaking what works I would prefer to leave it but I think the question stands and is valid:  when is quoting needed

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go (hunk 2 above)
Comment (jcantrill) [reply pkg/generators/forwarding/fluentd/fluent_conf_test.go:957]: yes.  this should be the default.  I believe this is here for when the change was not part of the original plugin

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go (hunk 1 above)
Comment (jcantrill) [reply pkg/generators/forwarding/fluentd/fluent_conf_test.go:330]: I don't believe so.  we have examples of where this is flipped the other way but it is the same fields

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go (hunk 2 above)
Comment (jcantrill) [reply pkg/generators/forwarding/fluentd/fluent_conf_test.go:957]: @periklis looks as I was wrong and should actually check the default https://bugzilla.redhat.com/show_bug.cgi?id=1864397  https://github.com/uken/fluent-plugin-elasticsearch#sniffer-class-name

**Pull request merged**